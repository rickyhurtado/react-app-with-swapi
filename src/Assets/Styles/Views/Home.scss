.home-page {
  position: relative;
  text-align: center;
  &-header {
    padding: 20px;
    @media (max-width: 600px) {
      padding: 0;
    }
  }
  &-logo {
    height: 300px;
    @media (max-width: 600px) {
      height: auto;
      width: 100%;
    }
  }
  .page-title {
    font-size: 1.5em;
    font-weight: normal;
  }
  @media (max-width: 470px) {
    .nav-button {
      display: block;
      font-size: 25px;
      margin-top: 15px;
      width: inherit;
      &:first-child {
        margin: 0;
      }
    }
  }
}
