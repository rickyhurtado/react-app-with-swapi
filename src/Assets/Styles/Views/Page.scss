.page {
  &-logo {
    height: 150px;
    @media (max-width: 400px) {
      height: auto;
      width: 100%;
    }
  }
  &-title {
    font-size: 30px;
    font-weight: normal;
    margin-top: -20px;
    text-transform: uppercase;
    @media (max-width: 330px) {
      font-size: 25px;
      margin-top: -15px;
    }
  }
  .page-sub-title {
    font-size: 20px;
    font-weight: normal;
    text-transform: uppercase;
  }
  nav {
    .nav-button {
      border: 2px solid #fee91a;
      font-size: 20px;
      min-width: 100px;
      padding: 10px;
      @media (max-width: 330px) {
        font-size: 18px;
        min-width: 80px;
        padding: 7px;
      }
    }
    &.pagination {
      span.nav-button {
        border-color: #333;
        color: #333;
      }
    }
  }
}
