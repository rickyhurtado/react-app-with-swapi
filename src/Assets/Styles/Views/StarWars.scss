.star-wars {
  &-list {
    list-style: none;
    padding: 0;
    &-item {
      font-size: 20px;
      padding: 10px 0;
      @media (max-width: 420px) {
        font-size: 18px;
        padding: 7px 0;
      }
      a {
        &:hover {
          font-weight: bold;
        }
      }
      &-small {
        font-size: 16px;
        padding: 5px;
        @media (max-width: 420px) {
          font-size: 14px;
        }
      }
    }
  }
}

