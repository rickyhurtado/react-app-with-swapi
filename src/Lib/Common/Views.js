import React from 'react';
import { Link } from 'react-router-dom';
import { Helmet } from 'react-helmet';

export const PageTitle = ({ ...props }) => {
  let siteName = process.env.REACT_APP_SITE_NAME;
  let { title, customTitle, parent } = props;

  if (parent) {
    siteName = `${parent} | ${siteName}`;
  }

  let headTitle = customTitle ? title : `${title} | ${siteName}`;

  if (!title) {
    title = siteName;
    headTitle = siteName;
  }

  return (
    <h1 className="page-title">
      <Helmet title={headTitle} />
      {title}
    </h1>
  );
};

export const Navigations = ({...props}) => {
  return (
    <nav>
      <Link to="/" className="nav-button">Home</Link>
      <Link to={`/${props.resources}`} className="nav-button">{props.title}</Link>
    </nav>
  );
};
