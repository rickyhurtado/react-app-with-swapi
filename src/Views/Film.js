import React, { Component } from 'react';
import DataListItem from '../Components/DataListItem';

export default class Film extends Component {
  render() {
    return (
      <DataListItem
        title="Films"
        attribute="Characters"
        recordId={this.props.match ? this.props.match.params.id : null}
      />
    );
  }
};
