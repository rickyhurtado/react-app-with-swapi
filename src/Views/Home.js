import React, { Component } from 'react';
import { withRouter, Link } from 'react-router-dom';
import StarWarsLogo from '../Assets/Images/Star_Wars_Logo.png';
import { PageTitle } from '../Lib/Common/Views';

class Home extends Component {
  render() {
    return (
      <div className="home-page">
        <header className="home-page-header">
          <img src={StarWarsLogo} className="home-page-logo" alt="StarWars Logo" />
          <PageTitle title="React App with SWAPI" />
        </header>
        <nav>
          <Link to="/people" className="nav-button">People</Link>
          <Link to="/films" className="nav-button">Films</Link>
        </nav>
      </div>
    );
  }
};

export default withRouter(Home);
