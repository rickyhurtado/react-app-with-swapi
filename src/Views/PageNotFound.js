import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { PageTitle } from '../Lib/Common/Views';

export default class PageNotFound extends Component {
  render() {
    return (
      <div className="page-not-found-page">
        <PageTitle title="404" />
        <nav>
          <Link to="/" className="nav-button single">Home</Link>
        </nav>
      </div>
    );
  }
};
