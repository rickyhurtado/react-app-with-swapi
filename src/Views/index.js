import Home from './Home';
import Film from './Film';
import Films from './Films';
import PageNotFound from './PageNotFound';
import People from './People';

export default {
  Home,
  Film,
  Films,
  PageNotFound,
  People
}
