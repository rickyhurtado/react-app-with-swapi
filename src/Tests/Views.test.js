import React from 'react'
import { BrowserRouter, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Views from '../Views';

test('StarWars Views.Home is rendered', () => {
  const component = renderer.create(
    <BrowserRouter>
      <Route>
        <Views.Home />
      </Route>
    </BrowserRouter>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('StarWars Views.People is rendered', () => {
  const component = renderer.create(
    <Views.People />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('StarWars Views.Film is rendered', () => {
  const component = renderer.create(
    <Views.Film />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('StarWars Views.Films is rendered', () => {
  const component = renderer.create(
    <Views.Films />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('StarWars Views.PageNotFound is rendered', () => {
  const component = renderer.create(
    <BrowserRouter>
      <Route>
        <Views.PageNotFound />
      </Route>
    </BrowserRouter>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
