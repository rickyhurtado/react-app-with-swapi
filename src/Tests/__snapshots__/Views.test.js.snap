// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarWars Views.Film is rendered 1`] = `
<div
  className="data-list-item"
>
  <div>
    <h1
      className="page-title"
    >
      Films
    </h1>
    <p>
      Loading...
    </p>
  </div>
</div>
`;

exports[`StarWars Views.Films is rendered 1`] = `
<div>
  <h1
    className="page-title"
  >
    Films
  </h1>
  <p>
    Loading Star Wars Films...
  </p>
</div>
`;

exports[`StarWars Views.Home is rendered 1`] = `
<div
  className="home-page"
>
  <header
    className="home-page-header"
  >
    <img
      alt="StarWars Logo"
      className="home-page-logo"
      src="Star_Wars_Logo.png"
    />
    <h1
      className="page-title"
    >
      React App with SWAPI
    </h1>
  </header>
  <nav>
    <a
      className="nav-button"
      href="/people"
      onClick={[Function]}
    >
      People
    </a>
    <a
      className="nav-button"
      href="/films"
      onClick={[Function]}
    >
      Films
    </a>
  </nav>
</div>
`;

exports[`StarWars Views.PageNotFound is rendered 1`] = `
<div
  className="page-not-found-page"
>
  <h1
    className="page-title"
  >
    404
  </h1>
  <nav>
    <a
      className="nav-button single"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </nav>
</div>
`;

exports[`StarWars Views.People is rendered 1`] = `
<div>
  <h1
    className="page-title"
  >
    People
  </h1>
  <p>
    Loading Star Wars People...
  </p>
</div>
`;
